addons/webhook/certificates.go

```diff
@@ -0,0 +1,125 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	certutil "k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
+	rsrcs "knative.dev/pkg/webhook/certificates/resources"
+	"path"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"time"
+)
+
+const (
+	ServerKey = "server-key.pem"
+	// ServerCert is the name of the key associated with the secret's public key.
+	ServerCert = "server-cert.pem"
+	// CACert is the name of the key associated with the certificate of the CA for
+	// the keypair.
+	CACert = "ca-cert.pem"
+)
+
+func SetupWebhooks(mgr *manager.Manager, namespace string, vWhCfgNames, mWhCfgNames []string) error {
+
+	//create the secret and write the certificate files based on the existing webhook server config
+	wh := (*mgr).GetWebhookServer()
+	secret, err := UpdateTLSSecret("addons-webhook-tls", wh, namespace)
+	if err != nil {
+		return err
+	}
+
+	//update the webhooks with the new CA from the secret created
+	k8sConfig, err := ctrl.GetConfig()
+	if err != nil {
+		return err
+	}
+	client, err := kubernetes.NewForConfig(k8sConfig)
+	if err != nil {
+		return err
+	}
```

Comment:
It would be nice to have obtain the "client" (kubernetes.interface) from the manager itself, but I am not sure if that is possible.

Reply:
manager has the controller-runtime client and not the more generic kubernetes.interface client